connector/spanmetricsconnectorv2/connector.go

```diff
@@ -0,0 +1,85 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package spanmetricsconnectorv2 // import "github.com/elastic/opentelemetry-collector-components/connector/spanmetricsconnectorv2"
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const scopeName = "otelcol/spanmetricsconnectorv2"
+
+type spanMetrics struct {
+	component.StartFunc
+	component.ShutdownFunc
+
+	next            consumer.Metrics
+	spansMetricDefs map[string]metricDef
+}
+
+func (sm *spanMetrics) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+func (sm *spanMetrics) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	var multiError error
+	processedMetrics := pmetric.NewMetrics()
+	processedMetrics.ResourceMetrics().EnsureCapacity(td.ResourceSpans().Len())
+	for i := 0; i < td.ResourceSpans().Len(); i++ {
+		resourceSpan := td.ResourceSpans().At(i)
+		// TODO (lahsivjar): add support for exponential histogram and summary
+		spansHist := newExplicitHistogram(sm.spansMetricDefs)
```

Comment:
Is the idea to use exponential histograms by default once support for that has been added? I think that would be preferable over using explicit boundaries.

Reply:
I haven't thought about what should be the default but it makes sense to make exponential histograms as default.